### PR TITLE
Solution to use JsonIgnore on OpenAPI schema objects #263a

### PR DESCRIPTION
**Pull request by Dreamwalker666**

### CODE REVIEW

**Comment by Dreamwalker666** [comment]
Opps not sure how that a got on the end of the issue number.  This obviously should like to #263 :)

File: src/Carter/OpenApi/CarterOpenApi.cs [hunk 1]
```diff
@@ -368,9 +371,19 @@ private static void AddSchema(Dictionary<string, SchemaElement> navigation, Open
                 
                 foreach (var memberKeyValue in keyValuePair.Value.DataMembers)
                 {
-                    var propertySchema = SchemaFromElement(memberKeyValue.Value);
                     var propertyInfo = keyValuePair.Value.ElementType.GetProperties().Where(o => CamelCase(o.Name) == memberKeyValue.Key).SingleOrDefault();
                     object[] attribute = propertyInfo.GetCustomAttributes(typeof(ApiSchemaAttributes), true);
+
+                    if (options.SchemaIgnoreAttribute != null)
```
Comment (jchannon) [inline src/Carter/OpenApi/CarterOpenApi.cs:377]: Won't this always be not null?

File: src/Carter/CarterOptions.cs [hunk 2]
```diff
@@ -56,6 +56,13 @@ public class OpenApiOptions
         /// Determines if OpenAPI should be mapped and enabled. True by default.
         /// </summary>
         public bool Enabled { get; set; } = true;
+        
+        /// <summary>
+        /// Attribute used to ignore peoperties from the OpenAPI schema
```
Comment (jchannon) [inline src/Carter/CarterOptions.cs:61]: Typo 😁

File: src/Carter/OpenApi/CarterOpenApi.cs [hunk 3]
```diff
@@ -351,7 +351,10 @@ private static OpenApiDocument CreateDocument(CarterOptions options)
         /// </summary>
         /// <param name="navigation">The input SchemaElements that have been read from the input classes.</param>
         /// <param name="document">The OpenApiDocument that is to be filled with the schema information.</param>
-        private static void AddSchema(Dictionary<string, SchemaElement> navigation, OpenApiDocument document, HttpContext context)
+        /// <param name="context"></param>
```
Comment (jchannon) [inline src/Carter/OpenApi/CarterOpenApi.cs:354]: Can we fill these parameters info in?

**Review by jchannon — CHANGES_REQUESTED** [review]
Looks good! Thanks

File: src/Carter/OpenApi/CarterOpenApi.cs (hunk 1 above)
Comment (Dreamwalker666) [reply src/Carter/OpenApi/CarterOpenApi.cs:377]: You can set the option to null to turn off the feature.  I did it like that for backwards compatibility.  But thinking on it now who would use the attribute and expect it not to get ignored?

File: samples/CarterSample/Features/Actors/Actor.cs [hunk 4]
```diff
@@ -1,9 +1,14 @@
 namespace CarterSample.Features.Actors
 {
+    using System.Text.Json.Serialization;
```
Comment (JoeStead) [inline samples/CarterSample/Features/Actors/Actor.cs:3]: Should this be a carter-specific attribute instead? Given Carter can be configured with it's JSON Serialiser, it's possible to mix-and-match with it here. If using Newtonsoft this seems a bit odd to me

File: src/Carter/CarterOptions.cs [hunk 5]
```diff
@@ -56,6 +56,13 @@ public class OpenApiOptions
         /// Determines if OpenAPI should be mapped and enabled. True by default.
         /// </summary>
         public bool Enabled { get; set; } = true;
+        
+        /// <summary>
+        /// Attribute used to ignore properties from the OpenAPI schema
+        ///
+        /// Note: Set this to null to disable.
+        /// </summary>
+        public Type SchemaIgnoreAttribute { get; set; } = typeof(System.Text.Json.Serialization.JsonIgnoreAttribute);
```
Comment (JoeStead) [inline src/Carter/CarterOptions.cs:65]: Ahh, I see it's possible to override. Ignore the last comment :)

Comment (Dreamwalker666) [reply src/Carter/CarterOptions.cs:65]: Yeah didn't want to tie it to just the internal .net attribute.  Too many serialisers out there :)

**Comment by jchannon** [comment]
Thanks! 👍

**Pull request merged**